### PR TITLE
Pic 3588 unit tests

### PR DESCRIPTION
**Pull request by rschallapalli**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/probation/courtcaseservice/controller/HmppsDocumentManagementApiGatewayControllerIntTest.kt [hunk 1]
```diff
@@ -92,6 +92,21 @@ internal class HmppsDocumentManagementApiGatewayControllerIntTest: BaseIntTest()
...
+        RestAssured.given()
+            .multiPart("file", File("./src/test/resources/document-upload/test-upload-file.xyz"))
+            .auth()
+            .oauth2(TokenHelper.getToken())
+            .contentType(ContentType.MULTIPART)
+            .`when`()
+            .post("/hearing/{hearingId}/defendant/{defendantId}/file", HEARING_ID, DEFENDANT_ID)
+            .then()
+            .statusCode(415)
+            .body("userMessage", equalTo("Unsupported or missing file type {xyz}. Supported file types [csv, doc, docx, jpg, jpeg, xml, ods, odt, pdf, png, ppt, pptx, rdf, rtf, txt, xls, xlsx, xml, zip]"))
```
Comment (AndySweilam) [inline src/test/kotlin/uk/gov/justice/probation/courtcaseservice/controller/HmppsDocumentManagementApiGatewayControllerIntTest.kt:107]: Maybe put them as a variable somewhere to be able to change them later

File: src/main/kotlin/uk/gov/justice/probation/courtcaseservice/service/HmppsDocumentManagementService.kt [hunk 2]
```diff
@@ -45,24 +50,44 @@ class HmppsDocumentManagementService (val hmppsDocumentManagementApiClient: Hmpp
             ?: throw EntityNotFoundException("Document not found /hearing/%s/defendant/%s/documents/%s", hearingId, defendantId, documentId)
     }
 
-    fun getDocument(hearingId: String, defendantId: String, documentId: String): Optional<Mono<ResponseEntity<Flux<InputStreamResource>>>> {
+    fun getDocument(
+        hearingId: String,
+        defendantId: String,
+        documentId: String
+    ): Mono<ResponseEntity<Flux<InputStreamResource>>>? {
```
Comment (AndySweilam) [inline src/main/kotlin/uk/gov/justice/probation/courtcaseservice/service/HmppsDocumentManagementService.kt:57]: That mono is awfully complex as return type, can we do ResponseEntity<T> or maybe map it?

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/exceptions/UnsupportedFileTypeException.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package uk.gov.justice.probation.courtcaseservice.service.exceptions;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+
+public class UnsupportedFileTypeException extends RuntimeException {
+    public UnsupportedFileTypeException(String fileType, List<String> allowedExtensions) {
+        super(String.format("Unsupported or missing file type {%s}. Supported file types %s", fileType, StringUtils.join(allowedExtensions)));
+    }
```
Comment (AndySweilam) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/service/exceptions/UnsupportedFileTypeException.java:10]: nice

**Review by AndySweilam — DISMISSED** [review]

File: src/main/kotlin/uk/gov/justice/probation/courtcaseservice/service/HmppsDocumentManagementService.kt (hunk 2 above)
Comment (rschallapalli) [reply src/main/kotlin/uk/gov/justice/probation/courtcaseservice/service/HmppsDocumentManagementService.kt]: addressed, thanks

**Review by AndySweilam — APPROVED** [review]

**Pull request merged**